integrated_channels/integrated_channel/exporters/content_metadata.py

```diff
@@ -285,32 +285,25 @@ def _get_catalog_diff(
                     course_or_course_run_key=item.get('content_key')
                 )
 
-        # if we have more to work with than the allowed space, slice it up
-        if len(unique_new_items_to_create) + len(items_to_delete) + len(matched_items) > max_item_count:
```

Comment:
i suspected that the logic here was somehow dropping deletes as it was trying to re-size so, i moved this into a utility method with some test cases to prove it was working as expected.